src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,788 @@
+                lease.LastRenewal = DateTime.UtcNow;
+                lease.ExpiresAt = DateTime.UtcNow.Add(this.options.LeaseInterval);
+            }
+
+            void StealLease(TableLease lease)
+            {
+                lease.NextOwner = this.workerName;
+            }
+           
+            // Log operations on partition table.
+            void LogHelper(
+                TableLease partition,
+                bool claimedLease,
+                bool stoleLease,
+                bool renewedLease,
+                bool drainedLease,
+                bool releasedLease,
+                string previousOwner)
+            {
+                if (claimedLease)
+                {
+                    this.settings.Logger.LeaseAcquisitionSucceeded(
+                        this.storageAccountName,
+                        this.settings.TaskHubName,
+                        this.settings.WorkerId,
+                        partitionId: partition.RowKey,
+                        leaseType: NotApplicable);
+                }
+                if (stoleLease)
+                {
+                    this.settings.Logger.LeaseStealingSucceeded(
+                        this.storageAccountName,
+                        this.settings.TaskHubName,
+                        this.settings.WorkerId,
+                        previousOwner,
+                        leaseType: NotApplicable,
+                        partitionId: partition.RowKey);
+                }
+                if (releasedLease)
+                {
+                    this.settings.Logger.LeaseRemoved(
+                        this.storageAccountName,
+                        this.settings.TaskHubName,
+                        this.settings.WorkerId,
+                        partition.RowKey,
+                        token: NotApplicable,
+                        leaseType: NotApplicable);
+                }
+                if (drainedLease)
+                {
+                    this.settings.Logger.PartitionManagerInfo(
+                        this.storageAccountName,
+                        this.settings.TaskHubName,
+                        this.settings.WorkerId,
+                        partitionId: partition.RowKey,
+                        details: "Draining partition");
+                }
+                if (renewedLease)
+                {
+                    this.settings.Logger.LeaseRenewalResult(
+                        this.storageAccountName,
+                        this.settings.TaskHubName,
+                        this.settings.WorkerId,
+                        partitionId: partition.RowKey,
+                        success: true,
+                        token: NotApplicable,
+                        leaseType: NotApplicable,
+                        details: "Successfully renewed partition lease");
+                }
+            }
+
+            // Add any other active worker's lease to dictionary partitionDistribution for further balance process.
```

Comment:
nit - trying to improve clarity
```suggestion
            // Track partition ownership in the partitionDistribution dictionary. We use this information when rebalancing partitions.
```